azurelinuxagent/common/persist_firewall_rules.py

```diff
@@ -94,6 +100,10 @@ def _is_firewall_service_running():
         return False
 
     def setup(self):
+        if not CGroupsApi.is_systemd():
+            logger.warn("Did not detect Systemd, unable to set {0}".format(self._network_setup_service_name))
```

Comment:
logging as a warning instead of an error and moved one step up because checking firewall.service also requires systemd to be there